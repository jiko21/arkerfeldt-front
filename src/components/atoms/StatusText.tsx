/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { FC } from 'react';
import { Color } from '@/const/color';

const wrapperStyle = css`
  align-items: center;
  background-color: ${Color.LIGHT_GRAY};
  display: flex;
  justify-content: center;
  height: 24px;
  width: 80px;
`;

const spanStyle = css`
  color: ${Color.LIGHT_TEXT};
  font-size: 14px;
  font-weight: 700;
`;

const StatusText: FC = ({ children }) => (
  <div css={wrapperStyle}>
    <span css={spanStyle}>{children}</span>
  </div>
);

export default StatusText;
