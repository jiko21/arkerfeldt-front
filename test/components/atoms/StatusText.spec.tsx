import StatusText from '@/components/atoms/StatusText';
import { render } from '@testing-library/react';

describe('StatusText.tsx', () => {
  let onClickMock: jest.Mock;
  beforeEach(() => {
    onClickMock = jest.fn();
  });

  afterEach(() => {
    onClickMock.mockClear();
  });
  it('should correctly render', () => {
    const { container } = render(<StatusText>Sample</StatusText>);

    expect(container).toMatchSnapshot();
  });
});
