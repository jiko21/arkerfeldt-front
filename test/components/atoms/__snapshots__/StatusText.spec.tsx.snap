// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusText.tsx should correctly render 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #C4C4C4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 24px;
  width: 80px;
}

.emotion-1 {
  color: #6A6A6A;
  font-size: 14px;
  font-weight: 700;
}

<div>
  <div
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      Sample
    </span>
  </div>
</div>
`;
